subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -141,6 +141,54 @@ These errors can be resolved by updating the violating code or dependency. Updat
 
 For some examples of possible error or warning outputs which may arise due to this change, see <<remove_test_add_opens>>.
 
+==== SourceSet classesDirs no longer depends upon the entire SourceSet as a task dependency
+
+Prior to Gradle 8.0, the task dependencies for the `classesDirs` of a given `SourceSetOutput`
+included the `SourceSetOutput` itself as a dependency. This meant that directly
+computing `classesDirs` would also execute `classes`, `processResources`, and any other
+task dependency for the `SourceSetOutput`.
+Since 8.0, this implicit dependency is removed. Now, computing `classesDirs` only executes the
+tasks which directly contribute to it and none of the tasks which contribute to the parent `SourceSetOutput`.
```

Comment:
```suggestion
Prior to Gradle 8.0, the task dependencies for the `SourceSetOutput.classesDirs` 
included tasks that did not produce class files. This meant that a task that
depends on `classesDirs` would also depend on `classes`, `processResources`, and any other
task dependency added to `SourceSetOutput`. This behavior was potentially an error because the `classesDirs` property did not contain the output for `processResources`. 
Since 8.0, this implicit dependency is removed. Now, depending on `classesDirs` only executes the
tasks which directly produce files in the classes directories.
```